orchestrator/relayer/src/logic_call_relaying.rs

```diff
@@ -44,8 +46,12 @@ pub async fn relay_logic_calls(
         debug!("Got sigs {:?}", sigs);
         if let Ok(sigs) = sigs {
             let hash = encode_logic_call_confirm_hashed(gravity_id.clone(), call.clone());
-            // this checks that the signatures for the batch are actually possible to submit to the chain
+            // this checks that the signatures for the logic call are actually possible to submit to the chain
             if current_valset.order_sigs(&hash, &sigs).is_ok() {
+                if logic_call_skips.should_skip(&call) {
+                    continue;
+                }
```

Comment:
Wouldn't it be optimal to do the skip at the very beginning of this for loop?

Reply:
👍